### PR TITLE
Add dummy test for self hosted GitHub runners

### PR DESCRIPTION
**Pull request by tomassebestik**
This is simple test for newly installed self hosted RPI Github Runners in Espressif Brno site.
Before accepting this PR some selfhosted GH runner must be assigned into project.

Details in https://jira.espressif.com:8443/browse/RDT-43

### CODE REVIEW

File: .github/workflows/test_selfhosted_runner.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Test Github action on self hosted RPI runnes
+
+on:
+  push:
+    branches: [pull_request]
```
Comment (me-no-dev) [inline .github/workflows/test_selfhosted_runner.yml:5]: Shouldn't this be
```yml
on:
  push:
    branches:
    - master
  pull_request:
```

or

```yml
on:
  pull_request:
```

Comment (me-no-dev) [reply .github/workflows/test_selfhosted_runner.yml]: Now the test ran :) https://github.com/espressif/arduino-esp32/pull/5521/checks?check_run_id=3310070479

Could you sign the CLA too please? GH Checks are complaining

**Review by hippyaki — COMMENTED** [review]
What good does it do? Can you help me understand?

**Comment by me-no-dev** [comment]
> What good does it do? Can you help me understand?

this is a test for future use.

File: .github/workflows/test_selfhosted_runner.yml (hunk 1 above)
Comment (tomassebestik) [reply .github/workflows/test_selfhosted_runner.yml]: @me-no-dev Done, thank you. PLS merge it.

**Comment by jepler** [comment]
This causes CI in forks to fail, because they do not have such a runner.

**Pull request merged**